initfiles/bash/etc/init.d/hpcc_common.in

```diff
@@ -649,9 +649,14 @@ status_component() {
   check_status ${PIDPATH} ${LOCKPATH} ${COMPPIDPATH} 1
   RCSTATUS=$?
   getPid ${COMPPIDPATH}
-  if [ ${RCSTATUS} -ne 0 ];then
+  if [[ ${RCSTATUS} -ne 0 ]]; then
     log "${compName} ---> Stopped"
     printf "%-15s is stopped" "$compName"
+  elif [[ ${compType} == "thor" ]]; then
+    __slavenum=$(cat ${runtime}/${compName}/thorgroup | wc -l)
+    __hostnum=$(cat ${runtime}/${compName}/slaves | sort | uniq | wc -l)
```

Comment:
I can explicitly check if you think that is a better way to do it.  Its implied that they exist because the thormaster process has successfully started up